exporting patch:
# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1506353400 18000
#      Mon Sep 25 10:30:00 2017 -0500
# Node ID f3c4a7256d65d42baab6bc77dd51a8c0fb12eb06
# Parent  1265c7f0119a75356f1b4950e9a2f8ec819d81bc
Move worker thread shutdown to main window handling finish (bug #50025)

* main-window.cc (main_window::main_window): Remove octave_finished_signal()
  connection from m_interpreter--which runs in m_main_thread--to
  m_main_thread's quit().
  (main_window::handle_octave_finished): Quit and wait on the worker thread.
  Put some fprintfs in the code for debugging assistance.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- libgui/src/main-window.cc
+++ libgui/src/main-window.cc
@@ -217,9 +217,6 @@ main_window::main_window (QWidget *p, oc
   connect (m_interpreter, SIGNAL (octave_finished_signal (int)),
            this, SLOT (handle_octave_finished (int)));
 
-  connect (m_interpreter, SIGNAL (octave_finished_signal (int)),
-           m_main_thread, SLOT (quit (void)));
-
   connect (m_main_thread, SIGNAL (finished (void)),
            m_main_thread, SLOT (deleteLater (void)));
 
@@ -1579,7 +1576,14 @@ main_window::handle_octave_ready (void)
 void
 main_window::handle_octave_finished (int exit_status)
 {
+fprintf(stderr, "HANDLE_OCTAVE_FINISHED: 111111111111111111111111111111111111111111\n");
+  m_main_thread->quit();
+fprintf(stderr, "HANDLE_OCTAVE_FINISHED: 222222222222222222222222222222222222222222\n");
+  m_main_thread->wait();
+fprintf(stderr, "HANDLE_OCTAVE_FINISHED: 333333333333333333333333333333333333333333\n");
+
   qApp->exit (exit_status);
+fprintf(stderr, "HANDLE_OCTAVE_FINISHED: 444444444444444444444444444444444444444444\n");
 }
 
 void
